From 284e1291603a0c703fa78b6d30ba0c653c2a56b8 Mon Sep 17 00:00:00 2001
From: Tim Yao <tim.yao@amlogic.com>
Date: Wed, 4 Dec 2019 22:23:50 -0800
Subject: [PATCH] aml_libs_nf: fix build for library install [1/1]

PD#SWPL-17555

Problem:
Patchbuild failed because the target folder is not created

Solution:
Change install option to make target install working

Verify:
Local

Change-Id: I24429bfe355342b1af14a8b54998665ee4163ed5
Signed-off-by: Tim Yao <tim.yao@amlogic.com>
---
 src/amadec/Makefile                      | 2 +-
 src/amavutils/Makefile                   | 2 +-
 src/amcodec/Makefile                     | 2 +-
 src/audio_codec/libadpcm/Makefile        | 2 +-
 src/audio_codec/libamr/Makefile          | 2 +-
 src/audio_codec/libape/Makefile          | 2 +-
 src/audio_codec/libcook/Makefile         | 2 +-
 src/audio_codec/libfaad/Makefile         | 2 +-
 src/audio_codec/libflac/Makefile         | 2 +-
 src/audio_codec/liblpcm/Makefile         | 2 +-
 src/audio_codec/libmad/Makefile          | 2 +-
 src/audio_codec/libpcm/Makefile          | 2 +-
 src/audio_codec/libraac/Makefile         | 2 +-
 src/audio_codec/wfd_aac_decoder/Makefile | 2 +-
 14 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/src/amadec/Makefile b/src/amadec/Makefile
index ad12177..df312f2 100644
--- a/src/amadec/Makefile
+++ b/src/amadec/Makefile
@@ -31,7 +31,7 @@ $(TARGET): $(TARGET_OBJS)
 
 install:
 	-install -Dm 644 include/dtv_patch_out.h $(AML_LIBS_NF_STAGING_DIR)/usr/include/aml_libs_nf/amadec/dtv_patch_out.h
-	-install -m 555 ${TARGET} $(INSTALL_DIR)
+	-install -Dm 555 ${TARGET} $(INSTALL_DIR)/$(TARGET)
 
 force:
 
diff --git a/src/amavutils/Makefile b/src/amavutils/Makefile
index 1756b85..e8cbb94 100644
--- a/src/amavutils/Makefile
+++ b/src/amavutils/Makefile
@@ -32,7 +32,7 @@ $(TARGET): $(TARGET_OBJS)
     
 
 install:
-	-install -m 555 ${TARGET} $(INSTALL_DIR)
+	-install -Dm 555 ${TARGET} $(INSTALL_DIR)/${TARGET}
 force:
 
 clean:
diff --git a/src/amcodec/Makefile b/src/amcodec/Makefile
index bb46b15..0d300f6 100644
--- a/src/amcodec/Makefile
+++ b/src/amcodec/Makefile
@@ -59,4 +59,4 @@ clean:$(CLRDIR)
 	rm -rf $(target_all)
 
 install:$(target_all)
-	-install  -m 755   $(TARGET)  $(INSTALL_DIR)
+	-install  -Dm 755   $(TARGET)  $(INSTALL_DIR)/$(TARGET)
diff --git a/src/audio_codec/libadpcm/Makefile b/src/audio_codec/libadpcm/Makefile
index 6eb507c..86f33ea 100644
--- a/src/audio_codec/libadpcm/Makefile
+++ b/src/audio_codec/libadpcm/Makefile
@@ -29,7 +29,7 @@ $(TARGET): $(TARGET_OBJS)
     
 
 install:
-	-install -m 555 ${TARGET} $(INSTALL_DIR)
+	-install -Dm 555 ${TARGET} $(INSTALL_DIR)/${TARGET}
 
 force:
 
diff --git a/src/audio_codec/libamr/Makefile b/src/audio_codec/libamr/Makefile
index 6a25235..5530a62 100644
--- a/src/audio_codec/libamr/Makefile
+++ b/src/audio_codec/libamr/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/libape/Makefile b/src/audio_codec/libape/Makefile
index 4f3125a..0186b28 100644
--- a/src/audio_codec/libape/Makefile
+++ b/src/audio_codec/libape/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/libcook/Makefile b/src/audio_codec/libcook/Makefile
index 3489a76..18d0a97 100644
--- a/src/audio_codec/libcook/Makefile
+++ b/src/audio_codec/libcook/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/libfaad/Makefile b/src/audio_codec/libfaad/Makefile
index 6ed7b0a..5c4d63d 100644
--- a/src/audio_codec/libfaad/Makefile
+++ b/src/audio_codec/libfaad/Makefile
@@ -81,7 +81,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean default
 
diff --git a/src/audio_codec/libflac/Makefile b/src/audio_codec/libflac/Makefile
index ffb8425..5d8fd39 100644
--- a/src/audio_codec/libflac/Makefile
+++ b/src/audio_codec/libflac/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/liblpcm/Makefile b/src/audio_codec/liblpcm/Makefile
index adc30af..ad3c061 100644
--- a/src/audio_codec/liblpcm/Makefile
+++ b/src/audio_codec/liblpcm/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/libmad/Makefile b/src/audio_codec/libmad/Makefile
index 07fed8c..f7a2a59 100644
--- a/src/audio_codec/libmad/Makefile
+++ b/src/audio_codec/libmad/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/libpcm/Makefile b/src/audio_codec/libpcm/Makefile
index 412d7cb..a9c6ebc 100644
--- a/src/audio_codec/libpcm/Makefile
+++ b/src/audio_codec/libpcm/Makefile
@@ -75,7 +75,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/libraac/Makefile b/src/audio_codec/libraac/Makefile
index 4229cad..658d2c3 100644
--- a/src/audio_codec/libraac/Makefile
+++ b/src/audio_codec/libraac/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
diff --git a/src/audio_codec/wfd_aac_decoder/Makefile b/src/audio_codec/wfd_aac_decoder/Makefile
index 07fed8c..f7a2a59 100644
--- a/src/audio_codec/wfd_aac_decoder/Makefile
+++ b/src/audio_codec/wfd_aac_decoder/Makefile
@@ -73,7 +73,7 @@ $(OUTPUT):$(OBJS)
 -include $(DEPENDS)
 
 install:
-	-install -m 555 ${OUTPUT} $(INSTALL_DIR)
+	-install -Dm 555 ${OUTPUT} $(INSTALL_DIR)/${OUTPUT}
 	
 rebuild:clean all
 
-- 
2.7.4

